### ISSUE TITLE
Tests fail (nil value handling)

### ISSUE THREAD

**Issue by Knorkebrot** [ansel1/merry#4]
In 99536c2 you changed the nil value handling and added a test for this (`TestNilValues`). It contains the following line:

```go
if Wrap(nil).WithHTTPCode(400).WithMessage("hey") { //...
```

Maybe I misunderstand your commit, but you can't call methods on "true" nil interfaces, can you?

http://play.golang.org/p/mmJoXJR9J1

Did you mean

```go
if (WithMessage(WithHTTPCode(Wrap(nil), 400), "hey") != nil) { //...
```

(This will still crash since `WithValue` doesn't handle the nil from the `Wrap` call it does.)

**Comment by Knorkebrot** [comment]
https://github.com/Knorkebrot/merry/commit/df92772c9acca8f2f3a71ed542d7f523f6359558

**Comment by ansel1** [comment]
Fixed with #6
